src/components/FilterGroup.jsx

```diff
@@ -0,0 +1,34 @@
+import React, { useCallback } from "react";
+
+import { and, isEmpty } from "ramda";
+
+import FilterBlock from "./FilterBlock";
+import FilterDropdown from "./FilterDropdown";
+import JoinSelector from "./JoinSelector";
+
+const FilterGroup = ({ conditions }) => {
+  const shouldDisplayJoinSelector = useCallback(
+    (index) => and(!isEmpty(conditions), index !== conditions.length - 1),
```

Comment:
Infix form is easier to comprehend.
```suggestion
    (index) => !isEmpty(conditions) && index !== conditions.length - 1,
```